components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp

```diff
@@ -536,17 +537,17 @@ function displayPasswordField() {
     <td align="left">
     	 <input type="hidden" id="isSystem" name="isSystem" value="<%=isSystem %>"/>
     	 <%if (editMode) { %>
-            <input id="dsName" name=dsName class="longInput" value="<%=dataSourceName %>" readonly>
+            <input id="dsName" name=dsName class="longInput" value="<%=Encode.forHtml(dataSourceName) %>" readonly>
         <%} else { %>
-        	<input id="dsName" name=dsName class="longInput" value="<%=dataSourceName %>">
+        	<input id="dsName" name=dsName class="longInput" value="<%=Encode.forHtml(dataSourceName) %>">
         <%} %>
         <input type="hidden" id="editMode" name="editMode" value="<%=editMode %>"/>
     </td>
 </tr>
 <tr>
     <td style="width:170px;"><fmt:message key="description"/></td>
     <td align="left">
-        <input id="description" name="description" class="longInput" value="<%=description %>" />
+        <input id="description" name="description" class="longInput" value="<%=Encode.forHtml(description) %>" />
```

Comment:
can the description value be null? if so, wouldn't this throw an exception?
can we check for all the encoding values here?